### PR TITLE
Fix issue with `allow_mismatch: false` flag

### PR DESCRIPTION
**Pull request by sbbroot**
* An exception will be raised when `allow_mismatch` flag is set to `false`
  while downloading images

### CODE REVIEW

File: ansible/playbooks/roles/repository/files/download-requirements/src/mode/base_mode.py [hunk 1]
```diff
@@ -131,21 +161,33 @@ def __download_files(self, files: Dict[str, Dict], dest: Path):
...
 
-                logging.info(f'- {file}')
-                self._download_file(file, filepath)
+                if not self.__is_checksum_valid(files, req_file, tmpfile, 'sha256'):
+                    tmpfile.unlink()
+                    raise ChecksumMismatch(f'- {req_file}')
 
-                if files[file]['sha256'] != get_sha256(filepath):
-                    raise ChecksumMismatch(f'- {file}')
+                move(str(tmpfile), str(filepath))
```
Comment (to-bar) [inline ansible/playbooks/roles/repository/files/download-requirements/src/mode/base_mode.py]: Can we avoid moving from tmp by default to not slow down?

**Review by przemyslavic — APPROVED** [review]

**Review by seriva — APPROVED** [review]

**Review by erzetpe — APPROVED** [review]

**Review by rafzei — APPROVED** [review]

**Pull request merged**